src/lib/Client/EzRecommendationClientInterface.php

```diff
@@ -24,14 +24,14 @@ interface EzRecommendationClientInterface
     /**
      * @return \EzSystems\EzRecommendationClient\Client\EzRecommendationClientInterface
      */
-    public function setCustomerId(int $customerId): self;
+    public function setCustomerId(?int $customerId = null): self;
 
     public function getCustomerId(): ?int;
 
     /**
      * @return \EzSystems\EzRecommendationClient\Client\EzRecommendationClientInterface
      */
-    public function setLicenseKey(string $licenseKey): self;
+    public function setLicenseKey(?string $licenseKey = null): self;
```

Comment:
I noticed that `licenseKey` and `customerId` can now be null. In which situation is this valid?

If it's the case, then I assume that they are either both filled or both empty, so this method can be merged into one.

Second, I believe that providing a default value should not happen in this case. If `setLicense()` method is to be called with the intent of removing it, `null` value should be explicitly passed by the user.

Reply:
I changed this to be consistent with getters which signature are marked that they can return null: https://github.com/ezsystems/ezrecommendation-client/blob/master/src/lib/Client/EzRecommendationClientInterface.php#L27-L36. 

Answering on your question, `customerId` and `licenseKey` can be null e.g.: just right after install DXP or when customer doesn’t want to use the Personalization module. 

I’ld to open different PR where I change default settings for this bundle which set env's as optional. From 4.X customer can add more then one configurations and overwrite this settings by semantic configuration.:   

```yaml
parameters:
    ezrecommendation.default.authentication.customer_id: '%env(default::int:RECOMMENDATION_CUSTOMER_ID)%'
    ezrecommendation.default.authentication.license_key: '%env(default::RECOMMENDATION_LICENSE_KEY)%'
    ezrecommendation.default.included_content_types: []
    ezrecommendation.default.host_uri: '%env(default::RECOMMENDATION_HOST_URI)%'
```